### PR TITLE
helm: add annotations to daemonset and deployment

### PR DESCRIPTION
**Pull request by cyril-corbon**
Signed-off-by: Cyril Corbon <cyril.corbon@dailymotion.com>

helm: add possibility to set annotations for deployment and daemonset

### CODE REVIEW

**Review by marquiz — COMMENTED** [review]
Thanks @cyril-corbon. Could you also document these new options in the table(s) in`docs/get-started/deployment-and-usage.md`?
/ok-to-test

**Comment by marquiz** [comment]
Any comments @eliaskoromilas @adrianchiris?

File: deployment/helm/node-feature-discovery/templates/master.yaml [hunk 1]
```diff
@@ -5,6 +5,10 @@ metadata:
   labels:
     {{- include "node-feature-discovery.labels" . | nindent 4 }}
     role: master
+{{- with .Values.master.deploymentAnnotations }}
+  annotations:
+    {{- toYaml . | nindent 4 }}
+{{- end }}
```
Comment (marquiz) [inline deployment/helm/node-feature-discovery/templates/master.yaml:11]: How about the simpler form we use below for pod annotations?

```suggestion
  annotations:
    {{- toYaml .Values.master.deploymentAnnotations | nindent 4 }}
```

File: deployment/helm/node-feature-discovery/templates/worker.yaml [hunk 2]
```diff
@@ -5,6 +5,10 @@ metadata:
   labels:
     {{- include "node-feature-discovery.labels" . | nindent 4 }}
     role: worker
+{{- with .Values.worker.daemonsetAnnotations }}
+  annotations:
+    {{- toYaml . | nindent 4 }}
+{{- end }}
```
Comment (marquiz) [inline deployment/helm/node-feature-discovery/templates/worker.yaml:11]:
```suggestion
  annotations:
    {{- toYaml .Values.worker.daemonsetAnnotations | nindent 4 }}
```

**Comment by marquiz** [comment]
Sorry about nitting, but the commit message looks cryptic to me, how about e.g. 
`helm: add annotations to daemonset and deployment`

File: deployment/helm/node-feature-discovery/templates/worker.yaml (hunk 2 above)
Comment (cyril-corbon) [reply deployment/helm/node-feature-discovery/templates/worker.yaml:11]: As you want, the `with` will just not print the annotations if the dict is empty

Comment (marquiz) [reply deployment/helm/node-feature-discovery/templates/worker.yaml:11]: I just find the template easier to read this way. And, I don't think the output really matters as it's normally "machine interpreted" anyway

**Comment by cyril-corbon** [comment]
review's applied

**Review by marquiz — APPROVED** [review]
Thanks @cyril-corbon for the contribution! Looks good to me. Let's give some time for others  to chime in, in case they have something...

**Comment by marquiz** [comment]
No comments from anybody else. IMO this is rather uncontroversial (and limited in scope) so...
/lgtm

**Comment by eliaskoromilas** [comment]
Looks good to me, too. Just out of curiosity, @cyril-corbon what's your use case?

**Comment by cyril-corbon** [comment]
we are using [reloader](https://github.com/stakater/Reloader) to auto update our workload when a configmap / secret is updated

**Comment by marquiz** [comment]
Hmm, github issues, again 🧐 (?)
/lgtm

**Pull request merged**